src/main/java/universe/UniverseAnnotatedTypeFactory.java

```diff
@@ -155,92 +149,6 @@ public UniversePropagationTreeAnnotator(AnnotatedTypeFactory atypeFactory) {
-                                    ? annos
-                                    : atypeFactory
-                                            .getQualifierHierarchy()
-                                            .leastUpperBounds(prev, annos);
-                }
-            } else {
-                prev = componentType.getAnnotations();
-            }
-
-            assert prev != null
-                    : "PropagationTreeAnnotator.visitNewArray: violated assumption about"
-                            + " qualifiers";
-
-            Pair<Tree, AnnotatedTypeMirror> context =
-                    atypeFactory.getVisitorState().getAssignmentContext();
-            Collection<? extends AnnotationMirror> post;
-
-            if (context != null
-                    && context.second != null
-                    && context.second instanceof AnnotatedTypeMirror.AnnotatedArrayType) {
-                AnnotatedTypeMirror contextComponentType =
-                        ((AnnotatedTypeMirror.AnnotatedArrayType) context.second)
-                                .getComponentType();
-                // Only compare the qualifiers that existed in the array type
-                // Defaulting wasn't performed yet, so prev might have fewer qualifiers than
-                // contextComponentType, which would cause a failure.
-                // TODO: better solution?
-                boolean prevIsSubtype = true;
-                for (AnnotationMirror am : prev) {
-                    if (contextComponentType.isAnnotatedInHierarchy(am)
-                            && !atypeFactory
-                                    .getQualifierHierarchy()
-                                    .isSubtype(
-                                            am,
-                                            contextComponentType.getAnnotationInHierarchy(am))) {
-                        prevIsSubtype = false;
-                    }
-                }
-                // TODO: checking conformance of component kinds is a basic sanity check
-                // It fails for array initializer expressions. Those should be handled nicer.
-                if (contextComponentType.getKind() == componentType.getKind()
-                        && (prev.isEmpty()
-                                || (!contextComponentType.getAnnotations().isEmpty()
-                                        && prevIsSubtype))) {
-                    post = contextComponentType.getAnnotations();
-                } else {
-                    // The type of the array initializers is incompatible with the
-                    // context type!
-                    // Somebody else will complain.
-                    post = prev;
-                }
-            } else {
-                // No context is available - simply use what we have.
-                post = prev;
-            }
-
-            // Below line is the only difference from super implementation
-            applyImmutableIfImplicitlyBottom(componentType);
```

Comment:
Can you add a test case that uses an "implicitly bottom" (whatever that was...) component type in an array to see whether this code actually did anything?

Reply:
@zcai1 I think you're looking into how to properly implement this. Should we merge this PR and you file an issue/open a new PR with test case and fixes?

Reply:
@zcai1 I think you're looking into how to properly implement this. Should we merge this PR and you file an issue/open a new PR with test case and fixes?

Reply:
@wmdietl yes, I think we should fix this in a separate PR.